app/adapters/alma_adapter.rb

```diff
@@ -48,9 +48,44 @@ def get_items_for_bib(id)
     bib_item_set.holding_summary
   end
 
+  # @param id [String]. e.g id = "991227850000541"
+  # @return [String] date record was created e.g. "2019-10-18Z"
+  def get_catalog_date(id)
+    record = get_bib_record(id)
+    get_catalog_date_from_record(record)
+  end
+
   private
 
     def apikey
       Rails.configuration.alma[:read_only_apikey]
     end
+
+    # @param record [AlmaAdapter::MarcRecord]
+    # @return [String] date record was created e.g. "2019-10-18Z"
+    def get_catalog_date_from_record(record)
+      return nil if record.nil?
+      ava = record.select { |f| f.tag == "AVA" }
+      if ava.count > 0
+        # Get the creation date from the physical items
+        dates = []
+        get_items_for_bib(record.bib.id).each do |_location, holdings|
+          items = holdings.map { |h| h["items"] }.compact.flatten
+          dates += items.map { |i| i["creation_date"] }.compact.flatten
```

Comment:
So, in https://github.com/pulibrary/marc_liberation/pull/1104 I removed a bunch of keys from getting returned by `get_items_for_bib`, and now creation date will never be in there. The test still passes because the bib record and the item record have the same creation date, so it falls back to the bib and it's right. Maybe we can get together and figure out how to adjust the code. I was worried this exact thing would happen when I made that change -- at that time nothing else was calling get_items_for_bib, but I wondered "what if someone comes along and wants to use it?"...

Reply:
I reintroduced the creation_date as a field that we return (since we already had it) and this is not an issue anymore.